### DISCUSSION TITLE
Dynamic size for gradient image?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#2574] [Help] [answered]
All my images: https://imgur.com/a/aKCQdie

What I want:
`magick input.png ( -clone 0 -define gradient:radii=4600,4600 radial-gradient:grey70-grey40 ) -compose DivideSrc -composite final.png`

What I must do, instead:
`magick input.png ( -size 2209x3089 -define gradient:radii=4600,4600 radial-gradient:grey70-grey40 ) -compose DivideSrc -composite final.png`

Why doesn't the first command show the same result as the second? Second works perfectly!

In case you're wondering, what's wrong with this guy? Well, the "-size" parameter bothers me. I want a way to use the size of first image in the stack and use it for the gradient image, instead of having to goddamning type it each time.

**Comment by snibgo** [comment, marked as answer]
In the first command, you have two images within the parentheses: the clone, and a gradient, so it doesn't work.

A solution is to use a user-defined variable for the size, where the variable is set from the first image. Like this:
```
magick in.png -set option:MYSIZE %[fx:w]x%[fx:h] ( -size %[MYSIZE] radial-gradient: ) -compose DivideSrc -composite out.png
```
If you use Windows BAT, you need to double the % signs.

We don't need the parentheses, so we can also remove the variable:
```
magick toes.png -size %[fx:w]x%[fx:h] radial-gradient: -compose DivideSrc -composite x.png
```
We can simplify further:
```
magick toes.png -size %G radial-gradient: -compose DivideSrc -composite x.png
```
For the escapes %h, %G etc, see https://imagemagick.org/script/escape.php

**Reply by ghost** [reply]
Snibgo, we don't deserve you!

**Reply by GeeMack** [reply]
We can set the size ahead of the parentheses with simply "-size WxH". That situation shouldn't require the "-set option:variable" even when creating the "gradient:" inside the parentheses. This command and output shows that it creates a 640x480 red canvas...

    magick logo: -size %[w]x%[h] ( xc:red ) info:

    logo:[0] GIF 640x480 640x480+0+0 8-bit sRGB 256c 28576B 0.000u 0:00.009
    xc:red[1] XC 640x480 640x480+0+0 16-bit sRGB 0.000u 0:00.005

**Reply by snibgo** [reply]
Yes. And we don't need the square brackets.
```
magick logo: -size %wx%h ( xc:red ) info:
```
There are many variations.
